<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        const p = new Promise((resolve,reject)=>{
            resolve('hello')
        })

        // 当链式调用时出现了错误，会穿透其中的链路，一直到最后的catch中进行处理
        p.then(value=>{
            throw 'err'
            console.log(111)
        }).then(value=>{
            console.log(222)
        }).then(value=>{
            console.log(333)
        }).catch(reason=>{
            console.warn(reason)
        })
    </script>
</body>
</html>